### ISSUE TITLE
DLL builds are linking the wrong libraries

### ISSUE THREAD

**Issue by JPGygax68** [ShiftMediaProject/FFmpeg#12]
This one goes back to the change that was made at my request, so I guess I have to bear part of the blame :-)

The building of the DebugDLL and ReleaseDLL versions of `avcodec`go awry because they try to link in `libavutil.lib` (`libavutild.lib` for debug), which are no longer import libraries but static libraries since the import libraries now follow the naming standard saying that import libraries (under Windows) should not have the `lib` prefix (precisely so that they can be told apart from static library builds).

**Comment by Sibras** [comment]
Does this occur in specific projects or all of them? I cannot reproduce this as all the projects provided in the repo have the correct "avutil(d).lib" set. (See https://github.com/ShiftMediaProject/FFmpeg/blob/master/SMP/libavcodec.vcxproj line 1487 you can see the DebugDLL is using avutild.lib not libavutild.lib). Try making sure you have the latest copy of the repo.

**Comment by JPGygax68** [comment]
This is the linker dependency line from the `DebugDLL` version of `libavdevice`:

`libavformatd.lib;libavcodecd.lib;libavutild.lib;libavfilterd.lib;Opengl32.lib;vfw32.lib;shlwapi.lib;strmiids.lib;%(AdditionalDependencies)`

However after explicitly reverting all my changes in TortoiseGit and doing another pull, the line is now as it should be.

Ok, now I understand what I did wrong. I copied, modified and used a project generation batch file, but DID NOT REBUILD `project_generate.exe`. Of course the previous version would still generate the `lib` prefixes.

Sorry, this was my mistake.
